### PR TITLE
Include *.json in the root files if they are specified as root

### LINKED ISSUE #33827: TSServer reports TS6307 in composite projects on JSON files included via include pattern
**TypeScript Version:**  3.6.3, actually most likely all versions since v3.2

**Search Terms:** composite, resolveJSONModule, tsserver, TS6307, include pattern

**Code**

Not particularly easy to show with code, but here's a repro:

https://github.com/lierdakil/ts6307-composite-json-tsserver-test

1. Clone
2. Run `npm install`
3. Run `node test.js` in a terminal
4. Observe output
5. Kill `test.js` (ctrl+C or whathaveyou)

Note: `test.js` just spawns `tsserver` and tells it to open `src/test.ts`.

Please see [README](https://github.com/lierdakil/ts6307-composite-json-tsserver-test#complete-listing) for a complete listing.

**Expected behavior:**

No error diagnostics.

**Actual behavior:**

In tsserver output, we see:

`{"seq":0,"type":"event","event":"configFileDiag","body":{"triggerFile":"src/test.ts","configFile":"tsconfig.json","diagnostics":[{"text":"File '/tmp/test/src/blabla.json' is not listed within the file list of project 'tsconfig.json'. Projects must list all files or use an 'include' pattern.","code":6307,"category":"error"}]}}`

That is, TS6307 on a JSON file, which is included in the project via `include` pattern.

**Additional observations:** 

- Issue only manifests when `composite: true`.
- Issue only manifests in `tsserver`, but not in `tsc` itself.
- Issue only manifests when `tsconfig.json` uses `include`, but not when it uses `files`.

**Related Issues:** 
#25636 #30785, also see PR #27844

**Other related issues (from external projects):**
https://github.com/TypeStrong/ts-loader/issues/905
https://github.com/TypeStrong/atom-typescript/issues/1512

### PR DESCRIPTION
**Pull request by sheetalkamat**
Fixes #33827

### CODE REVIEW

File: src/testRunner/unittests/tsserver/projectErrors.ts [hunk 1]
```diff
@@ -861,4 +861,92 @@ declare module '@custom/plugin' {
             ]);
         });
     });
+
+    describe("unittests:: tsserver:: Project Errors with resolveJsonModule", () => {
+        function createSessionForTest({ include }: { include: readonly string[]; }) {
+            const test: File = {
+                path: `${projectRoot}/src/test.ts`,
+                content: `import * as blabla from "./blabla.json"`
```
Comment (andrewbranch) [inline src/testRunner/unittests/tsserver/projectErrors.ts]: If you add a `console.log(blabla)`, you won’t have to enumerate the unrelated `_0_is_declared_but_its_value_is_never_read` diagnostic in each test, which might make the intent of the test more clear.

**Comment by amcasey** [comment]
Do we know why they used to be excluded?

**Comment by sheetalkamat** [comment]
> Do we know why they used to be excluded?

@RyanCavanaugh added it with project references but I didn't know that we excluded them. There is no reason to exclude them as far as I can see...

**Comment by amcasey** [comment]
@sheetalkamat Do we still want to exclude config files?  (Or does that somehow happen upstream?)

**Comment by sheetalkamat** [comment]
configs are never part of root file names.. (even through external projects as in that case we end up opening configured projects for those configs instead)

**Comment by pcowgill** [comment]
Is this fix live on npm yet?

**Comment by Bessonov** [comment]
@pcowgill  `3.8.0-dev.20191122` on npm fixes it for me.

**Comment by FlorianWendelborn** [comment]
To extend on @Bessonov’s fix, for me it was also necessary to force VSCode to use the workspace TypeScript version (duh).

Add this to your `.vscode/settings.json`:

```json
{
	"typescript.tsdk": "node_modules/typescript/lib"
}
```

Then, after a `reload`, VSCode finally understood `.json` imports again.

**Pull request merged**